assignments/lambda-classes.js

```diff
@@ -1 +1,121 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(personOptions) {
+      this.name = personOptions.name;
+      this.age = personOptions.age;
+      this.location = personOptions.location;
+      this.gender = personOptions.gender;
+    }
+    speak(){
+      return `Hello my name is ${this.name}, I am from ${this.location}.`
+    }
+  }
+  class Instructor extends Person {
+    constructor(instructorOptions) {
+      super(instructorOptions);
+      this.specialty = instructorOptions.specialty;
+      this.favLanguage = instructorOptions.favLanguage;
+      this.catchPhrase = instructorOptions.catchPhrase;
+    }
+    demo (subject){
+      return `Today we are learning about ${subject}.`;
+    }
+    grade (student, subject){
+      return `${student.name} receives a perfect score on ${subject}`;
+    }
+    teach(student){
+      if (student.grade >= 70){
+        return `${student.name}'s grade is ${student.grade} so ${student.name} is now a Lambda School graduate!`
+      }
+      else {
+        while (student.grade < 70){
+          student.grade = Math.floor(Math.random() * 100) + 1;
+          if (student.grade >= 70){
+            return `${student.name}'s grade is ${student.grade} so ${student.name} is now a Lambda School graduate!`
+          }
+          console.log(`${student.name}'s new grade is ${student.grade}`)
+        }
+      }
+    }
+  }
+  class Student extends Person {
+    constructor(studentOptions) {
+      super(studentOptions);
+      this.previousBackground = studentOptions.previousBackground;
+      this.className = studentOptions.className;
+      this.favSubjects = studentOptions.favSubjects;
+      this.grade = Math.floor(Math.random() * 100) + 1;  //stretch goal 
+    }
+    listsSubjects(){
+      for (var i = 1; i < 4; i++){
```

Comment:
great job with the for loop here, I would suggest you get away from using var and treat it like it doesn't exist. If you're not sure when to use const or let, then just go with let. Var can cause issues the const and let are created to stop.